src/test/scala/viper/server/core/ViperServerHttpSpec.scala

```diff
@@ -34,7 +34,9 @@ class ViperServerHttpSpec extends AnyWordSpec with Matchers with ScalatestRouteT
   private val viperServerHttp = {
     val config = new ViperConfig(IndexedSeq())
     val server = new ViperHttpServer(config)(verificationContext)
-    server.start()
+    val started = server.start()
+    // wait until server has been started:
+    Await.result(started, Duration.Inf)
```

Comment:
I would set a timeout to make the test fail if the it could not be started within some reasonable time, like 30 sec.